app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -102,17 +102,28 @@ class AuditController @Inject() (
       Future.successful(Accepted)
     }
 
-  private def buildDetails(payload: Payload)(implicit request: Request[Source[ByteString, _]]): EitherT[Future, PresentationError, Details] =
+  private def buildDetails(payload: Payload, auditSource: String)(implicit
+    request: Request[Source[ByteString, _]]
+  ): EitherT[Future, PresentationError, Details] =
     if (request.headers.get(Constants.XAuditMetaPath).isEmpty) {
       EitherT.leftT[Future, Details](PresentationError.badRequestError(s"${Constants.XAuditMetaPath} is missing"))
     } else {
+      var clientId                 = request.headers.get(Constants.XClientIdHeader).map(ClientId(_))
```

Comment:
This is the common code used in below section as well. Better to extract this in one method and then call that method here to avoid the code duplicacy